client/dashboard/sites/staging-site-sync-modal/index.tsx

```diff
@@ -174,9 +200,112 @@ export default function StagingSiteSyncModal( {
 		onClose();
 	}, [ onClose ] );
 
+	const { fileBrowserState } = useFileBrowserContext();
+	const browserCheckList = fileBrowserState.getCheckList();
+
+	const WP_CONFIG_PATH = '/wp-config.php';
+	const WP_CONTENT_PATH = '/wp-content';
+	const SQL_PATH = '/sql';
+
+	const wpContentNode = fileBrowserState.getNode( WP_CONTENT_PATH );
+	const wpConfigNode = fileBrowserState.getNode( WP_CONFIG_PATH );
+	const sqlNode = fileBrowserState.getNode( SQL_PATH );
+
+	const filesAndFoldersNodesCheckState = useMemo( () => {
+		const nodes = [ wpContentNode, wpConfigNode ].filter( Boolean );
+		if ( nodes.length === 0 ) {
+			return 'unchecked';
+		}
+		const checkedCount = nodes.filter( ( node ) => node?.checkState === 'checked' ).length;
+		const mixedCount = nodes.filter( ( node ) => node?.checkState === 'mixed' ).length;
+		if ( mixedCount > 0 ) {
+			return 'mixed';
+		}
+		if ( checkedCount === nodes.length ) {
+			return 'checked';
+		}
+		if ( checkedCount === 0 ) {
+			return 'unchecked';
+		}
+		return 'mixed';
+	}, [ wpContentNode, wpConfigNode ] );
+
+	const updateFilesAndFoldersCheckState = useCallback(
+		( checkState: 'checked' | 'unchecked' | 'mixed' ) => {
+			fileBrowserState.setNodeCheckState( WP_CONTENT_PATH, checkState );
+			fileBrowserState.setNodeCheckState( WP_CONFIG_PATH, checkState );
+		},
+		[ fileBrowserState ]
+	);
+
+	const onFilesFoldersCheckboxChange = useCallback( () => {
+		updateFilesAndFoldersCheckState(
+			filesAndFoldersNodesCheckState === 'checked' ? 'unchecked' : 'checked'
+		);
+	}, [ filesAndFoldersNodesCheckState, updateFilesAndFoldersCheckState ] );
+
+	const handleDatabaseCheckboxChange = useCallback( () => {
+		if ( sqlNode?.checkState === 'checked' ) {
+			fileBrowserState.setNodeCheckState( SQL_PATH, 'unchecked' );
+		} else {
+			fileBrowserState.setNodeCheckState( SQL_PATH, 'checked' );
+		}
+	}, [ fileBrowserState, sqlNode ] );
+
+	const handleFileSelectionModeChange = useCallback(
+		( value: string ) => {
+			const isExpanded = value === 'true';
+			setIsFileBrowserVisible( isExpanded );
+			if ( ! isExpanded ) {
+				updateFilesAndFoldersCheckState( 'unchecked' );
+			}
+		},
+		[ updateFilesAndFoldersCheckState ]
+	);
+
+	// Determine latest successful backup for rewindId/display
+	const activityQuery = useRewindableActivityLogQuery(
+		querySiteId,
+		{
+			name: SUCCESSFUL_BACKUP_ACTIVITIES,
+			aggregate: false,
+			number: 1,
+			sortOrder: 'desc',
+		},
+		{}
+	) as { data: BackupActivity[] | undefined; isLoading: boolean };
+	const { data: backupData, isLoading: isLoadingBackupAttempt } = activityQuery;
+	let lastKnownBackupAttempt: BackupActivity | undefined;
+	if ( Array.isArray( backupData ) && backupData.length > 0 ) {
+		lastKnownBackupAttempt = backupData.slice( 0, 1 ).shift();
+	} else {
+		lastKnownBackupAttempt = undefined;
+	}
```

Comment:
This could be simplified a bit, but it's just a matter of style, so up to you 😄 
```suggestion
	if ( Array.isArray( backupData ) ) {
		lastKnownBackupAttempt = backupData?.[0];
	} 
```

Reply:
Or even shorter 😄 
```
const lastKnownBackupAttempt = Array.isArray( backupData ) ? backupData?.[ 0 ] : undefined
```
But it is just a different way to write the same 😊

Reply:
Adjusted in https://github.com/Automattic/wp-calypso/pull/105794/commits/6a3f16ad66b3bf92b60882aa9222d7d364978cdf